scipy/stats/tests/test_stats.py

```diff
@@ -3982,12 +3995,22 @@ def testTwoVsThree(self):
         data1p = data1 + 0.01
         data1m = data1 - 0.01
         data2 = np.array([1.0, 2.0, 3.0])
-        self._testOne(data1p, data2, 'two-sided', 1.0 / 3, 1.0)
-        self._testOne(data1p, data2, 'greater', 1.0 / 3, 0.7)
-        self._testOne(data1p, data2, 'less', 1.0 / 3, 0.7)
-        self._testOne(data1m, data2, 'two-sided', 2.0 / 3, 0.6)
-        self._testOne(data1m, data2, 'greater', 2.0 / 3, 0.3)
-        self._testOne(data1m, data2, 'less', 0, 1.0)
+        self._testOne(data1p, data2, 'two-sided', 1.0 / 3, 1.0,
+                      attrs={'statistic_location': 2.01, 'statistic_sign': 1})
+        self._testOne(data1p, data2, 'greater', 1.0 / 3, 0.7,
+                      attrs={'statistic_location': 2.01, 'statistic_sign': 1})
+        self._testOne(data1p, data2, 'less', 1.0 / 3, 0.7,
+                      attrs={'statistic_location': 1, 'statistic_sign': -1})
+        self._testOne(data1m, data2, 'two-sided', 2.0 / 3, 0.6,
+                      attrs={'statistic_location': 1.99, 'statistic_sign': 1})
+        self._testOne(data1m, data2, 'greater', 2.0 / 3, 0.3,
+                      attrs={'statistic_location': 1.99, 'statistic_sign': 1})
+        # The distributions never cross, so their max distance is zero which is
+        # true beyond both edges of the union (in practice, 0.99 and 3.0 are
+        # both correct for "statistic_location"). Therefore, we do not check
+        # for statistic_location to avoid future false positive test failures.
+        self._testOne(data1m, data2, 'less', 0, 1.0,
+                      attrs={'statistic_sign': -1})
```

Comment:
I'm not sure this is the strongest test. If you already thought through each one and reasoned out the answer (rather than just relying on the output of `ks_2samp` to tell you what the reference values should be), we can leave it.

Personally, though, I'd prefer to construct a test that makes it obvious what the answers should be. For instance, perhaps we would construct two samples that correspond very closely except at one observation. 
```python3
x = np.arange(10, dtype=np.float64)
y = x.copy()
y[6] = 6.9
stats.ks_2samp(x, y)
```

Something similar could be done for `ks_1samp`.

Reply:
done now